apps/web-client/src/lib/db/orders/reconciliation.ts

```diff
@@ -0,0 +1,88 @@
+import type { DB, ReconciliationOrder } from "./types";
+
+export async function getAllReconciliationOrders(db: DB): Promise<ReconciliationOrder[]> {
+	const result = await db.execO<ReconciliationOrder>(
+		"SELECT id, supplier_order_ids, customer_order_line_ids, created FROM reconciliation_order ORDER BY id ASC;"
+	);
+	return result;
+}
+
+export async function getReconciliationOrder(db: DB, id: number): Promise<ReconciliationOrder> {
+	const result = await db.execO<ReconciliationOrder>(
+		"SELECT id, supplier_order_ids, customer_order_line_ids, created FROM reconciliation_order WHERE id = ?;",
+		[id]
+	);
+	if (!result.length) {
+		throw new Error(`Reconciliation order with id ${id} not found`);
+	}
+	return result[0];
+}
+
+export async function createReconciliationOrder(db: DB, supplierOrderIds: string[]) {
+	if (!supplierOrderIds.length) {
+		throw new Error("Reconciliation order must be based on at least one supplier order");
+	}
+
+	const supplierOrderIdsJoined = supplierOrderIds.join(", ");
+
+	const tt = await db.execA(`INSERT INTO reconciliation_order (supplier_order_ids) VALUES (?) RETURNING id;`, [supplierOrderIdsJoined]);
+	return tt[0][0];
+}
+/**
+ * The way I see it while the isbns are being scanned they'll be cached
```

Comment:
What's the improvement from debouncing ISBN additions to a reconciliation order? I can see the added complexity, but it's not clear to me what other advantages are there.